### ISSUE TITLE
No taskbar icon

### ISSUE THREAD

**Issue by Rendi-boi** [flathub/org.davmail.DavMail#6]
DavMail runs fine when installed thorugh flatpak, but there is no icon in the taskbar and no apparent way to access the settings again without it.
OS - Solus

**Comment by wberrier** [comment]
It's not necessarily normal for a flatpak release.  Some desktop environments don't show the tray icon any longer without any additional extensions (ie: gnome3).

Is Solus using gnome3?

**Comment by Rendi-boi** [comment]
Solus main desktop environment is Budgie, which i also use. It is built on gnome or something like that (i am not sure on terminology).

**Comment by penyuan** [comment]
I can confirm this problem on Gnome 3.32.2. I've installed Gnome extensions (e.g. Top Icons) to enable to system tray, and other apps such as Nextcloud or Skype can show their tray icons no problem, *but not DavMail*. This means I need to keep the DavMail window open all the time otherwise I can never regain access to it, it is a bit annoying!

Can this be fixed? I'd be happy to provide more diagnostic information to make the tray icon show up. Thanks!

**Comment by woook** [comment]
Same here - gnome in pop!_os

**Comment by penyuan** [comment]
Yeah this is still an ongoing problem. Can the maintainers and/or developers let us users know how to help diagnose the fix it?

**Comment by wberrier** [comment]
I've created a test build in #9 .  Mind trying it out?

The only issue I have with this change is that DavMail shows either the main window or the systray icon.

Gnome by default doesn't show systray icons.  This means for a vanilla install the user won't be presented with any gui elements to know that the program is running.

I think this could be solved by allowing the main window and the systray icon (most other apps seem to do this), but that'd take additional work.

Another option could be to show the main window by default, but then add a command line or property option to force showing the tray (currently there exists an option to *not* show the systray).

**Comment by wberrier** [comment]
Actually, never mind.

davmail will show the window if the extension isn't enabled, and will enable the tray icon if the extension is enabled (at least with the "Tray Icons" extension).

Please test and report back any issues with this test build.

**Comment by wberrier** [comment]
Issues (running on Fedora 32):

* when animating for activity, some icons don't look right
* sometimes the right click context menu is all over the place (not sure if that's a davmail bug or not?)
* even when running in the sys-tray, alt-tab still lists davmail for the workspace it was started on

These are probably all out of the scope of this change though...

**Comment by penyuan** [comment]
Sorry for double-posting, I'm not sure the best place for it so here's my report again:

Thank you @wberrier for the clear instructions, I have installed and test this test build. Here are my findings:

1. DavMail now starts directly into the system tray icon and is accessible from the icon! :+1:  This works fine in Gnome Wayland mode, *but* in Xorg mode the tray icon is completely invisible! I can click on and interact with the icon, but it is not visible...
2. In Xorg mode, there is a strange, tiny "window" that shows up even though I can't interact with it. Here is a GIF screen shot of me hovering my mouse cursor over that window in Gnome's window overview mode:

![](https://pic.infini.fr/BM8w2lOU/Ny4opnzI)

What does this mean???

I will keep this test build installed for now (it is functioning normally) to help test, just let me know what to do!

**Comment by mguessan** [comment]
Working fine under wayland and X11, closing old ticket
